src/components/singleProductPage/singleProductDetails.scss

```diff
@@ -0,0 +1,376 @@
+  }
+
+  .commentBtn {
+    background-color: transparent;
+    grid-row: 11/14;
+    grid-column: 6/12;
+    margin: 10%;
+    margin-right: 0;
+    width: 100%;
+    height: 30px;
+    border: none;
+    border-bottom: solid 2px #c4c4c490;
+    border-left: solid 2px #c4c4c490;
+  }
+}
+
+@include media(desktop) {
+  .productImgs {
+    .slick-thumb li {
+      margin: 15%;
+      margin-bottom: 30%;
+      margin-left: 0;
+    }
+  }
+}
+
+@include media(tablet) {
+  .productImgs {
+    .slick-thumb li {
+      margin: 15%;
+      margin-bottom: 40%;
+      margin-left: 5%;
+    }
+  }
+
+  .productInfo {
+    .productQuantity {
+      width: 50%;
+    }
+
+    .quantity {
+      grid-row: 4;
+      grid-column: 2/3;
+      font-size: 100%;
+    }
+  }
+}
+
+@include media(tablet-sm) {
+  .productInfoContainer {
+    width: 100%;
+    height: 50vh;
+  }
+
+  .productImgs {
+    width: 50%;
+
+    .smallImgs {
+      width: 220%;
+    }
+
+    .slick-thumb li {
+      margin: 15%;
+      margin-bottom: 50%;
+      margin-left: 0;
+    }
+  }
+
+  .productInfo {
+    display: grid;
+    grid-template-columns: repeat(12, 1fr);
+    grid-template-rows: repeat(12, 1fr);
+
+    .productBrand {
+      font-size: 17px;
+
+      &::before {
+        top: 50%;
+        left: -40%;
+      }
+    }
+
+    .productPrice {
+      font-size: 17px;
+    }
+
+    .productTitle {
+      grid-row: 2;
+      grid-column: 1/2;
+      font-size: 20px;
+    }
+
+    .productDescription {
+      grid-column: 1/12;
+      font-size: 13px;
+    }
+
+    .productQuantity {
+      grid-row: 4/6;
+      grid-column: 1/4;
+      width: 40%;
+      height: 60%;
+    }
+
+    .quantity {
+      grid-row: 4;
+      grid-column: 2;
+      font-size: 90%;
+    }
+
+    .productSizes {
+      width: 90%;
+      height: 60%;
+      font-size: 90%;
+    }
+
+    .addToCartBtn {
+      width: 100%;
+      height: 40%;
+      font-size: 70%;
+    }
+
+    .buyNowBtn {
+      width: 100%;
+      height: 40%;
+      font-size: 70%;
+    }
+
+    .favoriteBtn,
+    .commentBtn {
+      width: 100%;
+      height: 30px;
+    }
+  }
+}
+
+@include media(phone) {
+  .productInfoContainer {
+    height: 40vh;
+  }
+
+  .productInfo {
+    height: 90%;
+
+    .addToCartBtn {
+      width: 100%;
+      height: 50%;
+    }
+
+    .buyNowBtn {
+      width: 100%;
+      height: 50%;
+    }
+
+    .quantity {
+      font-size: 70%;
+      margin-left: -25%;
+    }
+  }
+}
+
+@include media(phone-sm) {
+  .productInfoContainer {
+    display: flex;
+    flex-direction: column;
+    height: 70vh;
+  }
+
+  .productInfo {
+    margin: 2%;
+
+    .productQuantity {
+      height: 30%;
+    }
+
+    .quantity {
+      font-size: 80%;
+      padding-top: 15%;
+    }
+
+    .productSizes {
+      height: 40%;
+      font-size: 80%;
+    }
+
+    .addToCartBtn {
+      width: 100%;
+      height: 30%;
+      grid-row: 5/8;
+    }
+
+    .buyNowBtn {
+      width: 100%;
```

Comment:
in general it's pretty strange that you have to have so much css. usually for responsive 99 percent of your css should "just work" for mobile displays without having to do media queries. in the future please be more aware of this and try to write your css so it magically works for smaller screens, otherwise this amount of code is really hard to maintain (it's hard to review css for that reason I'm not suggesting any concrete changes)

it may be also because of the usage of grid but it's not clear.

cc @emrerdem1